src/rebar_utils.erl

```diff
@@ -414,8 +414,14 @@ user_agent() ->
     ?FMT("Rebar/~s (OTP/~s)", [Vsn, otp_release()]).
 
 reread_config(ConfigList) ->
+    SetEnv = case version_tuple(?MODULE:otp_release()) of
+        {X, _, _} when X =< 17 ->
+            fun application:set_env/3;
+        _ ->
+            fun (App, Key, Val) -> application:set_env(App, Key, Val, [{persistent, true}]) end
+    end,
```

Comment:
I thought about function_exported, obviously, but it was exported in 17pre, it just got changed from (App, Key, Val, **Timeout**) to (App, Key, Val, **Options**). That's the least ugly solution that was available here, really.
As for persistence, the goal was again to mimic real sys.config. Normally, reloading an application doesn't trash sys.config.